### PR TITLE
Adding build scripts page.

### PR DESCRIPTION
**Pull request by maccoda**
Hopefully fulfills what #978 is looking for in regards to build scripts. Let me know if anything needs reworking 😄

### CODE REVIEW

File: src/cargo/build_scripts.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Build Scripts
+
+Sometimes a normal build from cargo is not enough. Perhaps your crate needs some
+pre-requites before cargo will successfully compile, things like code
```
Comment (marioidival) [inline src/cargo/build_scripts.md]: Fix typo in `pre-requites`

Comment (marioidival) [inline src/cargo/build_scripts.md]: Fix typo in `pre-requites`

Comment (maccoda) [reply src/cargo/build_scripts.md]: Whoops!

**Comment by marioidival** [comment]
@maccoda thank you

**Pull request merged**